src/odemis/driver/xt_client.py

```
@@ -2419,13 +2437,43 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
+                            delta = abs(current_pos[axis] - target_pos[axis])
+                            if delta > tol_rotation_failure:
+                                failures[axis] = (delta, tol_rotation_failure)
+
+                        if failures:
+                            msgs = [
+                                f"{axis}: error={delta:.6f} > tol={tol:.6f}"
+                                for axis, (delta, tol) in failures.items()
+                            ]
+                            raise CancelledError(
+                                "Stage move failure. Off-target axes:\n" + "\n".join(msgs)
+                            )
```

Ruff (0.14.5)

Abstract `raise` to an inner function

(TRY301)